### PR TITLE
build: Generate release notes for 2.2 release

### PR DESCRIPTION
**Pull request by tylerbutler**
Generated the release notes for 2.2 using the tool in https://github.com/microsoft/FluidFramework/pull/21951. No manual edits were made to the release note output.

To preview the release notes, I recommend viewing them directly on my PR branch here: https://github.com/tylerbutler/FluidFramework/blob/relnotes-2.2/RELEASE_NOTES/2.2.0.md

That ensures you see the formatting as it will be for customers.

### CODE REVIEW

**Comment by tylerbutler** [comment]
@CraigMacomber @yann-achard-MS @pragya91 Mentioning you since you have authored changesets for the release already. To preview what your notes will look like, see the PR description.

File: RELEASE_NOTES/2.2.0.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+Commit: [`25e74f9`](https://github.com/microsoft/FluidFramework/commit/25e74f9f3bed6e6ff041c088813c4cc1ea276b9c)
+
+Affected packages:
+
+- @fluidframework/tree
+- fluid-framework
+
+### ✨ New! `Record`-typed objects can now be used to construct MapNodes ([#22042](https://github.com/microsoft/FluidFramework/issues/22042))
+
+IYou can now construct MapNodes from `Record` typed objects, similar to how maps are expressed in JSON.
```
Comment (alexvy86) [inline RELEASE_NOTES/2.2.0.md:72]:
```suggestion
You can now construct MapNodes from `Record` typed objects, similar to how maps are expressed in JSON.
```

File: RELEASE_NOTES/2.2.0.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+Commit: [`25deff3`](https://github.com/microsoft/FluidFramework/commit/25deff344b447380486c1efb64ed69177c32ddc5)
+
+Affected packages:
+
+- fluid-framework
+- @fluidframework/tree
+
+### Implicit TreeNode construction improvements ([#21995](https://github.com/microsoft/FluidFramework/issues/21995))
+
+ArrayNodes and MapNodes could always be explicitly constructed (using `new`) from iterables. The types also allowed using of iterables to construct implicitly construct array nodes and map nodes, but this did not work at runtime. This has been fixed for all cases except implicitly constructing an ArrayNode form an `Iterable` that is actually a `Map`, and implicitly constructing a MapNode from an `Iterable` that is actually an `Array`. These cases may be fixed in the future, but require additional work to ensure unions of array nodes and map nodes work correctly.
```
Comment (alexvy86) [inline RELEASE_NOTES/2.2.0.md:98]:
```suggestion
ArrayNodes and MapNodes could always be explicitly constructed (using `new`) from iterables. The types also allowed using of iterables to implicitly construct array nodes and map nodes, but this did not work at runtime. This has been fixed for all cases except implicitly constructing an ArrayNode form an `Iterable` that is actually a `Map`, and implicitly constructing a MapNode from an `Iterable` that is actually an `Array`. These cases may be fixed in the future, but require additional work to ensure unions of array nodes and map nodes work correctly.
```

File: RELEASE_NOTES/2.2.0.md [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+#### Change details
+
+Commit: [`977f96c`](https://github.com/microsoft/FluidFramework/commit/977f96c1a0dd1d5eb0dbcd087d07cb7510d533ea)
+
+Affected packages:
+
+- fluid-framework
+- @fluidframework/tree
+
+### Fix document-corrupting bug when rebasing over move compositions. ([#21993](https://github.com/microsoft/FluidFramework/issues/21993))
```
Comment (alexvy86) [inline RELEASE_NOTES/2.2.0.md:111]:
```suggestion
### Fix document-corrupting bug when rebasing over move compositions. ([#21993](https://github.com/microsoft/FluidFramework/issues/21993))
```
For consistency, no trailing full stop on headers.
```suggestion
### Fix document-corrupting bug when rebasing over move compositions ([#21993](https://github.com/microsoft/FluidFramework/issues/21993))
```

**Review by alexvy86 — COMMENTED** [review]
I like how they look :)

File: .changeset/orange-ads-join.md [hunk 4]
```diff
@@ -2,10 +2,20 @@
 "@fluidframework/container-runtime": minor
 "@fluidframework/runtime-definitions": minor
 ---
+---
+section: deprecation
+---
+
+gcThrowOnTombstoneUsage and gcTombstoneEnforcementAllowed are deprecated
```
Comment (tylerbutler) [inline .changeset/orange-ads-join.md:9]: @agarwal-navin These are the changeset changes I made. To preview the release notes, I recommend viewing them directly on my PR branch here: https://github.com/tylerbutler/FluidFramework/blob/relnotes-2.2/RELEASE_NOTES/2.2.0.md

File: .changeset/fruity-birds-love.md [hunk 5]
```diff
@@ -5,9 +5,9 @@
 ---
 "section": "tree"
 ---
-Enable compile time type narrowing based on a TreeNode's NodeKind.
+Compile-time type narrowing based on a TreeNode's NodeKind
 
-TreeNode's schema aware APIs implement WithType, which now has a NodeKind parameter that can be used to narrow TreeNodes based on NodeKind.
+`TreeNode`'s schema aware APIs implement `WithType`, which now has a `NodeKind` parameter that can be used to narrow `TreeNode`s based on `NodeKind`.
```
Comment (jzaffiro) [inline .changeset/fruity-birds-love.md:10]:
```suggestion
`TreeNode`'s schema-aware APIs implement `WithType`, which now has a `NodeKind` parameter that can be used to narrow `TreeNode`s based on `NodeKind`.
```

File: .changeset/strong-mice-talk.md [hunk 6]
```diff
@@ -2,6 +2,9 @@
 "fluid-framework": minor
 "@fluidframework/tree": minor
 ---
+---
+includeInReleaseNotes: false
+---
 
 Add a function `isRepoSuperset` to determine if changes to a document schema are backward-compatible.
```
Comment (jzaffiro) [inline .changeset/strong-mice-talk.md:9]: nit: period at the end of line 9 should be removed

File: .changeset/sweet-things-wash.md [hunk 7]
```diff
@@ -1,6 +1,9 @@
 ---
 "@fluidframework/tree": minor
 ---
+---
+section: tree
+---
 
 Add `@alpha` API `FixRecursiveArraySchema` as a workaround around an issue with recursive ArrayNode schema.
```
Comment (jzaffiro) [inline .changeset/sweet-things-wash.md:8]: nit: period at the end of line 8 should be removed

**Review by jzaffiro — APPROVED** [review]
A few nits, but looks good overall!

**Pull request merged**